backend/baseclass.pm

```diff
@@ -1349,6 +1349,13 @@ sub stop_ssh_serial {
     return $self->{serial} = undef;
 }
 
+sub hide_password
+{
+    my ($self, %args) = @_;
+    $args{password} = 'SECRET' if ($args{password});
+    return %args;
```

Comment:
So now we copy `%args` not only once when passing them to `log_call` but also when 2 more times when passing/returning to/from `hide_password`. That doesn't seem efficient. (Not sure whether copy elision is a thing in Perl.)

Reply:
What do you propose? I see the following:

- Don't log `%args` at all
- introduce `bmwqemu::log_call_no_secret($array_ref_with_secrets, %args)`
- Use hash ref in `hide_password()` to remove one copy
- copy in each function and pass this modified copy to `log_call()`
- Create argument string by my self (or with pp) and pass it to log_call

In general I  don't see it critical to copy a array with ~10 entries, if this isn't call in a loop. But maybe I'm wrong.

Reply:
I think we need at least one copy because `hide_password` should only affect what's being logged. So we can only get rid of one copy:

```
bmwqemu::log_call(%{$self->hide_password(%args)});
```

So at least `hide_password` would return a reference.